src/walk.rs

```diff
@@ -289,6 +289,21 @@ pub fn scan(path_vec: &[PathBuf], pattern: Arc<Regex>, config: Arc<FdOptions>) {
                 }
             }
 
+            // Filter out unwanted modification times
+            if !config.modification_constraints.is_empty() {
+                if_chain!{
```

Comment:
I wanted to avoid the 4 * `else { return Continue }`.
I removed the dependency and added a boolean flag instead.
I could put back the else { return } if you prefer.